tests/prague/eip7692_eof_v1/eip7620_eof_create/test_subcontainer_validation.py

```diff
@@ -218,6 +222,101 @@ def test_container_combos_invalid(
                 code_section,
                 first_sub_container,
             ],
+            kind=container_kind,
+        ),
+        expect_exception=EOFException.INCOMPATIBLE_CONTAINER_KIND,
+    )
+
+
+@pytest.mark.parametrize(
+    "code_section,first_sub_container,container_kind",
+    [
+        pytest.param(
+            eofcreate_code_section,
+            stop_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/STOP",
+        ),
+        pytest.param(
+            eofcreate_code_section,
+            return_sub_container,
+            ContainerKind.RUNTIME,
+            id="EOFCREATE/RETURN",
+        ),
+        pytest.param(
+            returncontract_code_section,
+            returncontract_sub_container,
+            ContainerKind.INITCODE,
+            id="RETURNCONTRACT/RETURNCONTRACT",
+        ),
+    ],
+)
+def test_container_combos_deeply_nested_invalid(
+    eof_test: EOFTestFiller,
+    code_section: Section,
+    first_sub_container: Container,
+    container_kind: ContainerKind,
+):
+    """Test invalid subcontainer reference / opcode combos on a deep container nesting level"""
+    invalid_container = Container(
+        sections=[
+            code_section,
+            first_sub_container,
+        ],
+        kind=container_kind,
+    )
+
+    container = invalid_container
+    while len(container) < 0x6000:
+        container = Container(
+            sections=[
+                eofcreate_code_section,
+                Section.Container(container=container.copy()),
+            ],
+            kind=ContainerKind.INITCODE,
+        )
```

Comment:
```suggestion
        container = Container(
            sections=[
                eofcreate_code_section,
                Section.Container(
                    container=Container(
                        sections=[
                            Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
                            Section.Container(container=container.copy()),
                        ],
                    )
                ),
            ],
            kind=ContainerKind.INITCODE,
        )
```



I think this pattern is invalid in itself because all nested containers do EOFCREATE but only the deepest container does return contract.

IMO we should have a valid combination in the parametrize to prove that the pattern works if the deepest container is correct.

Reply:
It's not completely invalid, because EOFCREATE inside EOFCREATE initcode is allowed. And so I skipped RETURNCONTRACTs to make it somewhat simpler. However such initcode cannot end with STOP, so this was invalid for this reason, and I added now a variant that does REVERT after EOFCREATE, this is valid.

I've added positive cases too, and discovered an issue with how `container.copy()` works